liteclient/client.go

```diff
@@ -65,6 +65,32 @@ func GenerateAddresses(seed string, num int) []Address {
 // PrepareTransaction receives inputs and outputs and returns a signed transaction
 // inputsBody and outputsBody are JSONified arrays of TransactionInput and TransactionOutput, respectively.
 func PrepareTransaction(inputsBody string, outputsBody string) string {
+	newTransaction := buildTransaction(inputsBody, outputsBody, nil)
+	d := newTransaction.Serialize()
+
+	return hex.EncodeToString(d)
+}
+
+// PrepareTransactionWithSignatures receives inputs, outputs,and the signatures and returns.
+// inputsBody and outputsBody are JSONified arrays of TransactionInput and TransactionOutput, respectively.
+// signatureList is a JSONified array of strings.
+func PrepareTransactionWithSignatures(inputsBody string, outputsBody string, signatureList string) string {
+	var signatures []string
+	if err := json.Unmarshal([]byte(signatureList), &signatures); err != nil {
+		panic(err)
+	}
+
+	newTransaction := buildTransaction(inputsBody, outputsBody, signatures)
+	d := newTransaction.Serialize()
+
+	return hex.EncodeToString(d)
+}
+
+// Creates a coin.Transaction using the given lists of inputs, outputs and signatures. If signatureList is nill, the
+// signatures are created using the Secret property of each input.
+// inputsBody and outputsBody are JSONified arrays of TransactionInput and TransactionOutput, respectively.
+// signatureList is a JSONified array of strings.
```

Comment:
`signatureList is a JSONified array of strings.` its parsed at this point isn't it?

Reply:
Yes, it is. Changed